utils/frame/remote-externalities/src/lib.rs

```diff
@@ -501,16 +505,53 @@ impl<B: BlockT> Builder<B> {
 	pub async fn build(self) -> Result<TestExternalities, &'static str> {
 		let kv = self.pre_build().await?;
 		let mut ext = TestExternalities::new_empty();
+		let (offchain, _offchain_state) = TestOffchainExt::new();
+
+		// Register externality extensions in order to provide host interface for OCW to the runtime
+		ext.register_extension(OffchainDbExt::new(offchain.clone()));
+		ext.register_extension(OffchainWorkerExt::new(offchain));
 
 		info!(target: LOG_TARGET, "injecting a total of {} keys", kv.len());
 		for (k, v) in kv {
 			let (k, v) = (k.0, v.0);
+			// Insert the key,value pair into the test trie backend
 			ext.insert(k, v);
 		}
+
 		Ok(ext)
 	}
 }
 
+/// WS RPC API for one RPC calls to a substrate node.
+pub mod rpc_api {
```

Comment:
this RPC stuff is really all over the place... Ideally, I want to aggregate all of them in one utility crate in substrate's `utils`. You can see that I have similar code now in https://github.com/paritytech/polkadot/pull/3141/files as well. Something that we can work on in the long or near term.

Reply:
created issue: https://github.com/paritytech/substrate/issues/8988#issue-908681094